pkg/setting/setting.go

```diff
@@ -923,6 +928,18 @@ func (cfg *Cfg) readLDAPConfig() {
 	cfg.LDAPAllowSignup = LDAPAllowSignup
 }
 
+func (cfg *Cfg) readAWSConfig() {
+	awsPluginSec := cfg.Raw.Section("aws")
+	cfg.AWSAssumeRoleEnabled = awsPluginSec.Key("assume_role_enabled").MustBool(false)
+	allowedAuthProviders := awsPluginSec.Key("allowed_auth_providers").String()
+	for _, user := range strings.Split(allowedAuthProviders, ",") {
+		user = strings.TrimSpace(user)
+		if user != "" {
+			cfg.AWSAllowedAuthProviders = append(cfg.AWSAllowedAuthProviders, user)
```

Comment:
Could you validate the provider name here, to catch misconfigurations?

Reply:
hmm no I don't think we should. the point of having a list of providers is that they should be validated where they are implemented. in most cases, that will be the [grafana-aws-sdk](https://github.com/grafana/grafana-aws-sdk)

Reply:
@sunker ah, so we delegate the validation there? My main concern is that validation happens, not really where.

Reply:
I know the allowed providers are stated in the configuration doc, but we have to take in mind that external plugins might have different release cadence than grafana